include/tss2/tss2_esys.h

```diff
@@ -3302,6 +3404,17 @@ Esys_GetSysContext(
     ESYS_CONTEXT *esys_context,
     TSS2_SYS_CONTEXT **sys_context);
 
+TSS2_RC
+Esys_SetCryptoCallbacks(
+    ESYS_CONTEXT *esysContext,
+    ESYS_CRYPTO_CALLBACKS *callbacks,
+    void *userdata);
```

Comment:
a userdata pointer per callback makes the pointer struct very large.
IMHO a single one is nicer

Reply:
Thats what I was leaning towards as well. But refreshing myself we have ESYS_CRYPTO_CONTEXT_BLOB which is opaque, so that's really our user data here. I think we can drop that and make everything a `void *`. The callback `ESYS_CRYPTO_PK_RSA_ENCRYPT_FNP` however, doesn't take a crypto blob, so we should update that to include the `void *`.

Reply:
The purpose of the ESYS__CRYPTO_CONTEXT_BLOB was to keep track of the start/update/finish sequence info, right ?
Userdata would be more stable I guess.
So IMHO it is more convenient for the user to have these two things seperated.

Reply:
So do we want:
1. opaque struct type for the CONTEXT BLOB 
2. `void *` for userdata
    - Update all functions to take userdata and pass that userdata set in the SetCryptoCallbacks to all routines?